chalice/pipeline.py

```diff
@@ -45,12 +38,14 @@ class CreatePipelineTemplate(object):
         "Outputs": {},
     }
 
-    def __init__(self):
-        # type: () -> None
-        pass
+    def __init__(self, codebuild_image=None):
```

Comment:
What was the particular reason for adding the `codebuild_image` to the initializer instead of the `create_template()` method?  At first glance, I would assume the `codebuild_image` would be provided to the `create_template()` method and from there you would validate that both `codebuild_image` and `lambda_python_version` were not provided. By putting it in the initializer, an instance of the `CreatePipelineTemplate` could not be reused to build different pipelines for different images.

Reply:
Given I'm expecting to add more configuration options for generating your pipeline (e.g separate buildspec.yml, github source) I'm thinking about creating a `PipelineConfig` object you can pass to `create_template` to customize this process.  That way that object can own the validation.  It would also make it a little easier to add new options in the future.